app/packages/klogs/src/components/page/LogsPage.tsx

```diff
@@ -0,0 +1,233 @@
+      } else {
+        timeEnd = now();
+        timeStart = now() - timeOptions[search.time].seconds;
+      }
+
+      const order = orderMapping[search.order];
+      const path = `/api/plugins/klogs/logs?query=${search.query}&order=${order}&orderBy=${search.orderBy}&timeStart=${timeStart}&timeEnd=${timeEnd}`;
+      return client.get<ILogsData>(path, {
+        headers: {
+          'x-kobs-cluster': instance.cluster,
+          'x-kobs-plugin': instance.name,
+        },
+      });
+    },
+  );
+
+  const handleSearch = (query: string) => {
+    return setSearch({ fields: search.fields, query: query });
+  };
+
+  const handleChangeTime = ({ time, timeEnd, timeStart }: ITimes) => {
+    setLastSeach(now());
+    setSearch({
+      time: time,
+      ...(time === 'custom' ? { timeEnd, timeStart } : undefined),
+    });
+  };
+
+  const handleChangeTimeframe = (payload: IChangeTimeframePayload) => {
+    setSearch({ page: 1, time: 'custom', timeEnd: payload.timeEnd, timeStart: payload.timeStart });
+  };
+
+  const handleAddFilter = (filter: string) => {
+    const parts = [];
+    if (search.query) {
+      parts.push(search.query);
+      parts.push('_and_');
+    }
+    parts.push(filter);
+    handleSearch(parts.join(' '));
+  };
+
+  const handleChangeSort = (orderBy: string) => {
+    const isAsc = search.orderBy === orderBy && search.order === 'asc';
+    setSearch({
+      order: isAsc ? 'desc' : 'asc',
+      orderBy: orderBy,
+    });
+  };
+
+  const handleFieldToggle = (field: string) => {
+    const i = search.fields.indexOf(field);
+
+    if (i === -1) {
+      return setSearch({ fields: [...search.fields, field] });
+    }
+
+    return setSearch({ fields: [...search.fields.slice(0, i), ...search.fields.slice(i + 1)] });
+  };
+
+  const handleFieldSwap = (from: number, to: number) => {
+    setSearch({
+      fields: [
+        ...search.fields.slice(0, from),
+        search.fields[to],
+        ...search.fields.slice(from + 1, to),
+        search.fields[from],
+        ...search.fields.slice(to + 1),
+      ],
+    });
+  };
+
+  return (
+    <Page
+      title="klogs"
+      description="Fast, scalable and reliable logging using Fluent Bit and ClickHouse."
+      subtitle={instance.cluster}
+      toolbar={
+        <LogsToolbar
+          {...search}
+          handlers={{ onChangeTime: handleChangeTime, onSearch: handleSearch }}
+          instance={instance}
+        />
+      }
+      actions={
+        <>
+          <Tooltip title="Aggregation view">
+            <IconButton component={Link} to={`./aggregation${rawSearch}`}>
+              <PieChart />
+            </IconButton>
+          </Tooltip>
+          <Tooltip title="Documentation">
+            <IconButton component="a" href="https://kobs.io/main/plugins/klogs/" target="_blank">
+              <Description />
+            </IconButton>
+          </Tooltip>
+        </>
+      }
+    >
+      <UseQueryWrapper
+        {...queryResult}
```

Comment:
What's wrong with spread syntax for react components?

It will be difficult to keep up with your preferred coding style, if we don't add linting rules for things like this.